### PR TITLE
Added the junit test cases for CustomAuthorizationInterceptor

### PR DESCRIPTION
**Pull request by vsinghht**
**Overview**

- Added the tests for CustomAuthorizationInterceptor.

**How it was tested**

- Tested by running the tests locally.

**Checklist**

- [x] The title contains a short meaningful summary
- [x]  I have added a link to this PR in the Jira issue
- [x]  I have described how this was tested
- [ ]  I have included screen shots for changes that affect the user interface
- [ ]  I have updated unit tests
- [x]  I have run unit tests locally
- [ ]  I have updated documentation (including README)

### CODE REVIEW

**Review by shubhamparikh927 — APPROVED** [review]

File: src/main/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptor.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ private List<IAuthRule> unauthorizedRule() {
 			.build();
 	}
 
-	private List<IAuthRule> authorizeOAuth(RequestDetails theRequest) throws AuthenticationException {
+	protected List<IAuthRule> authorizeOAuth(RequestDetails theRequest) throws AuthenticationException {
```
Comment (hankwallace) [inline src/main/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptor.java]: No - tests shouldn't require changes to the underlying methods.

**Review by hankwallace — CHANGES_REQUESTED** [review]

**Review by hankwallace — CHANGES_REQUESTED** [review]
Did you review the built-in AuthorizationInterceptor tests to see how this should be implemented?

https://github.com/hapifhir/hapi-fhir/blob/fb3e141c4d7b790b824132b833c4bddc52e03835/hapi-fhir-jpaserver-test-r5/src/test/java/ca/uhn/fhir/jpa/provider/r5/AuthorizationInterceptorJpaR5Test.java

File: src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptorTest.java [hunk 2]
```diff
@@ -0,0 +1,708 @@
...
+	private static final String PATIENT_ID = "12345";
+	private static final String OBSERVATION_ID = "1234";
+	private static final String JWK_URL = "test-jwk-url";
+	private static final String ROLE_FHIR_ADMIN = "fhir4-admin";
+	private static final String ROLE_FHIR_USER = "fhir4-user";
+	private static final String API_KEY = "test-api-key";
+	private static final String ROLE_INVALID_USER = "invalid-user";
+
+	@BeforeEach
+	public void before() throws IOException {
```
Comment (hankwallace) [inline src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptorTest.java]: Why would a `before()` throw an exception?

**Review by hankwallace — CHANGES_REQUESTED** [review]
Did you review how the upstream repos implement interceptor tests?

Ex: https://github.com/hapifhir/hapi-fhir/blob/e2ca967fd987ceb506a75b85dd764895e0a98f14/hapi-fhir-server-openapi/src/test/java/ca/uhn/fhir/rest/openapi/OpenApiInterceptorWithAuthorizationInterceptorTest.java#L35

Please, do the following:
1. Remove the tests for API_KEY. We'll deprecate that very soon.
2. Separate the tests into different ones for remaining authorization methods.

File: src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptorTest.java [hunk 3]
```diff
@@ -0,0 +1,708 @@
...
+	private static Logger logger = LoggerFactory.getLogger(CustomAuthorizationInterceptorTest.class);
+	private static RestfulServer ourServlet;
+	private static boolean ourHitMethod;
+	private static List<Resource> ourReturn;
+	private static int ourPort;
+	private static CloseableHttpClient ourClient;
+	private static Server ourServer;
+	private static final FhirContext ourCtx = FhirContext.forR4();
+
+	private static final String TOKEN = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTUxNjIzOTAyMn0.NHVaYe26MbtOYhSKkoKYdFVomg4i8ZJd8_-RU8VNbftc4TSMb4bXP3l3YlNWACwyXPGffz5aXHc6lty1Y2t4SWRqGteragsVdZufDn5BlnJl9pdR_kdVFUsra2rWKEofkZeIC4yWytE58sMIihvo9H1ScmmVwBcQP6XETqYd0aSHp1gOa9RdUPDvoXQ5oqygTqVtxaDr6wUFKrKItgBMzWIdNZ6y7O9E0DhEPTbE9rfBo6KTFsHAZnMg4k68CDp2woYIaXbmYTWcvbzIuHO7_37GT79XdIwkm95QJ7hYC9RiwrV7mesbY4PAahERJawntho0my942XheVLmGwLMBkQ";
```
Comment (hankwallace) [inline src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptorTest.java:100]: Why is there a single hard-coded token?

Comment (vsinghht) [reply src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptorTest.java:100]: This is test token we have return below, as
 `DecodedJWT jwt = JWT.decode(token);` 
 method in CustomAuthorizationInterceptor requires a valid JWT to decode or we should mock this method as well.

**Pull request closed without merging**